eras/babbage/test-suite/src/Test/Cardano/Ledger/Babbage/Translation/TranslatableGen.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Test.Cardano.Ledger.Babbage.Translation.TranslatableGen (
+  genTx,
+  genTxOut,
+  utxoWithTx,
+) where
+
+import Cardano.Ledger.Babbage (Babbage, BabbageEra)
+import Cardano.Ledger.Babbage.Core
+import Cardano.Ledger.Crypto
+import Lens.Micro ((&), (.~), (^.))
+import Test.Cardano.Ledger.Babbage.Arbitrary ()
+import Test.Cardano.Ledger.Babbage.Serialisation.Generators ()
+
+import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.Alonzo.Scripts (AlonzoScript (..), ExUnits (..), Tag (..))
+import Cardano.Ledger.Alonzo.Scripts.Data (Data (..))
+import Cardano.Ledger.Alonzo.Tx (AlonzoTx (..))
+import Cardano.Ledger.Alonzo.TxWits
+import Cardano.Ledger.Babbage.TxBody (BabbageTxBody (..), BabbageTxOut (..), Datum (..))
+import Cardano.Ledger.Binary (mkSized)
+import Cardano.Ledger.Credential (StakeReference (..))
+import Cardano.Ledger.Language (Language (..))
+import Cardano.Ledger.TxIn (TxIn (..))
+import Cardano.Ledger.UTxO (UTxO (..))
+import qualified Data.Map as Map
+import Data.Maybe.Strict
+import Data.Sequence.Strict (fromList)
+import qualified Data.Set as Set
+import qualified PlutusLedgerApi.V1 as PV1
+import Test.Cardano.Ledger.Alonzo.Translation.TranslationInstanceGen (TranslatableGen (..))
+import Test.Cardano.Ledger.Core.Arbitrary ()
+import Test.QuickCheck (
+  Arbitrary,
+  Gen,
+  arbitrary,
+  frequency,
+  listOf1,
+  oneof,
+  scale,
+  vectorOf,
+ )
+
+instance TranslatableGen Babbage where
+  tgTx l = genTx @Babbage l (genTxBody l)
+  tgUtxo = utxoWithTx @Babbage
+
+utxoWithTx ::
+  forall era.
+  ( EraTx era
+  , Arbitrary (Value era)
+  , Arbitrary (Script era)
+  , BabbageEraTxBody era
+  , TxOut era ~ BabbageTxOut era
+  ) =>
+  Language ->
+  Tx era ->
+  Gen (UTxO era)
+utxoWithTx l tx = do
+  let ins = tx ^. bodyTxL ^. inputsTxBodyL
+  let refIns = tx ^. bodyTxL ^. referenceInputsTxBodyL
+  let allIns = ins `Set.union` refIns
+  outs <- vectorOf (length allIns) (genTxOut @era l)
```

Comment:
```suggestion
  let allIns = tx ^. bodyTxL ^. allInputsTxBodyL
  outs <- vectorOf (length allIns) (genTxOut @era l)
```

Reply:
Nice, this allowed me to get rid of this constraint : `BabbageEraTxBody era`